pkg/ip/ip_test.go

```diff
@@ -157,3 +158,35 @@ func TestRangeNetwork(t *testing.T) {
 		assert.EqualValues(t, n.Mask, te.n.Mask)
 	}
 }
+
+func TestRangeNetworkStringer(t *testing.T) {
+	symmetric := "sym"
+	var tests = []struct {
+		input  string
+		result string
+	}{
+		{"10.10.10.10/24", symmetric},
+		{"10.10.10.10-10.10.14.11", symmetric},
+		{"10.10.10.10-10.10.10.11", "10.10.10.10/31"},
+		{"10.10.10.128-10.10.10.129", "10.10.10.128/31"},
+		// test the boundaries of full subnets where the range is not the full subnet
+		{"10.10.10.125-10.10.10.127", "10.10.10.125/30"},
+		{"10.10.10.5-10.10.10.127", "10.10.10.5/25"},
+		{"10.10.10.0-10.10.10.127", "10.10.10.0/25"},
+		{"10.10.10.128-10.10.10.254", symmetric},
+		{"10.10.10.252-10.10.10.255", "10.10.10.252/30"},
+		// NOTE: I'm unsure why a single address translates into a range and not a /32
+		{"10.10.10.1-10.10.10.1", symmetric},
+		// {"10.10.10.2/32", symmetric},
```

Comment:
Is this an unneeded test case?

Reply:
It's there because I would expect, given the other ranges, to have the range collapsed to `/32`. It's not that it's needed but that it's illustrating something that's not occurring. I don't want to enshrine it as a negative test case because I think it should be allowed as it's consistent with the other range behaviours that end on an aligned address.